Updated to bevy 0.13

sclausen opened the pull request:
It seems `futures-lite` and `codespan-reporting` were unused dependencies, so I removed them.

AnthonyTornetta commented:
@sclausen Thanks for the PR! It worked well for me, except the naga_oil dependency breaks bevy when there's a 3d camera.
Error message (important bit at the top):
```
thread 'Async Compute Task Pool (1)' panicked at /home/cornchip/.cargo/registry/src/index.crates.io-6f17d22bba15001f/naga-0.19.2/src/back/glsl/mod.rs:1183:44:
internal error: entered unreachable code
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'Async Compute Task Pool (0)' panicked at /home/cornchip/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bevy_render-0.13.0/src/render_resource/pipeline_cache.rs:688:60:
called `Result::unwrap()` on an `Err` value: PoisonError { .. }
thread 'Async Compute Task Pool (1)' panicked at /home/cornchip/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bevy_render-0.13.0/src/render_resource/pipeline_cache.rs:688:60:
called `Result::unwrap()` on an `Err` value: PoisonError { .. }
thread 'Async Compute Task Pool (0)' panicked at /home/cornchip/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bevy_render-0.13.0/src/render_resource/pipeline_cache.rs:787:60:
called `Result::unwrap()` on an `Err` value: PoisonError { .. }
Encountered a panic in system `bevy_core_pipeline::upscaling::prepare_view_upscaling_pipelines`!
thread 'Compute Task Pool (1)' panicked at /home/cornchip/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bevy_render-0.13.0/src/pipelined_rendering.rs:49:67:
called `Result::unwrap()` on an `Err` value: RecvError
```

A fix I found is in the Cargo.toml, changing the `naga_oil = "0.13" to the same dependency options bevy has for their naga_oil dependency (https://github.com/bevyengine/bevy/blob/main/crates/bevy_render/Cargo.toml)

```toml
[target.'cfg(not(target_arch = "wasm32"))'.dependencies]
# Omit the `glsl` feature in non-WebAssembly by default.
naga_oil = { version = "0.13", default-features = false, features = [
  "test_shader",
] }
[target.'cfg(target_arch = "wasm32")'.dependencies]
naga_oil = "0.13"
```

sclausen commented:
@AnthonyTornetta exactly as the commit message says, I added a 3d camera to tone one_shot example and adjusted the Cargo.toml, as you suggested. Thank you very much for your input!

ivanceras reviewed (approved):
Works well.
